lib/internal/cluster/RoundRobinHandle.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+const assert = require('assert');
+const net = require('net');
+const { sendHelper } = require('internal/cluster/utils');
+const getOwnPropertyNames = Object.getOwnPropertyNames;
+const uv = process.binding('uv');
+
+module.exports = RoundRobinHandle;
+
+function RoundRobinHandle(key, address, port, addressType, fd) {
+  this.key = key;
+  this.all = {};
+  this.free = [];
+  this.handles = [];
+  this.handle = null;
+  this.server = net.createServer(assert.fail);
+
+  if (fd >= 0)
+    this.server.listen({ fd });
+  else if (port >= 0)
+    this.server.listen(port, address);
+  else
+    this.server.listen(address);  // UNIX socket path.
+
+  this.server.once('listening', () => {
+    this.handle = this.server._handle;
+    this.handle.onconnection = (err, handle) => this.distribute(err, handle);
+    this.server._handle = null;
+    this.server = null;
+  });
+}
+
+RoundRobinHandle.prototype.add = function(worker, send) {
+  assert(worker.id in this.all === false);
```

Comment:
Why not use `!` operator instead of comparison against false? As is done in

```
+  if (!(this instanceof Worker))
 +    return new Worker(options);
```